src/databricks/labs/ucx/installer/logs.py

```diff
@@ -0,0 +1,77 @@
+import dataclasses
+import logging
+import re
+from collections.abc import Iterator
+from pathlib import Path
+
+from databricks.labs.lsql.backends import SqlBackend
+
+from databricks.labs.ucx.framework.crawlers import CrawlerBase
+
+logger = logging.getLogger(__name__)
+
+
+@dataclasses.dataclass
+class LogRecord:
+    """A subset from logging.LogRecord.
+
+    Sources
+    -------
+    https://docs.python.org/3/library/logging.html#logging.LogRecord
+    """
+
+    level: int
+    msg: str
```

Comment:
```suggestion
    ts: int # fully specified timestamp in UTC (offset from a workspace fs file creation timestamp)
    job_id: int # determined from workspace file path, resolved from the install state
    workflow_name: str
    task_name: str
    job_run_id: int
    level: str # string, not int
    component: str
    message: str
```

Also link it in the installer here:
https://github.com/databrickslabs/ucx/blob/main/src/databricks/labs/ucx/install.py#L78-L95

Reply:
@nfx : How to find the `job_id` and `job_run_id`? 

[When creating the log files, these are fetched from the job parameters](https://github.com/databrickslabs/ucx/blob/aaf96a078886ce0562c646b95b8f57b6c3d332b3/src/databricks/labs/ucx/framework/tasks.py#L242). However, these parameters are only passed to the `TaskLogger` not the `Task` itself. I could reparse the `sys.args` in the logging task, which I would prefer to avoid to have one place of parsing arguments. This information is not available in `WorkspaceConfig` or `Installation`.